extensions/shared/external-media/media-browser/index.js

```diff
@@ -99,36 +104,92 @@ function MediaBrowser( props ) {
 		);
 	};
 
+	/**
+	 * Counts how many items are in a row by checking how many of the grid's child items have a matching offsetTop.
+	 */
+	const checkColumns = () => {
+		let perRow = 1;
+
+		const items = gridEl.current.children;
+
+		if ( items.length > 0 ) {
+			const firstOffset = items[ 0 ].offsetTop;
+
+			// Check how many items have a matching offsetTop.
+			// This will give us the total number of items in a row.
+			while ( perRow < items.length && items[ perRow ].offsetTop === firstOffset ) {
+				++perRow;
+			}
+		}
+
+		setColumns( perRow );
+	};
+
+	useEffect( () => {
+		checkColumns();
+	}, [ media ] );
```

Comment:
I think this check is good for setting the initial value. To prevent it from firing every time the media prop changes, we could do something like:

```js
useEffect( () => {
  if ( ! columns ) {
    checkColumns();
  }
}, [ media ] );
```

Reply:
In which case `columns` needs to be a dep of the effect right?

Reply:
Updated in https://github.com/Automattic/jetpack/pull/16118/commits/18e12bceb9c97fa17ae4f531cbbea638b6aae120. In that implementation columns value is set only once, when media are loaded.